File: python/mkuser.py

import splunklib.client as client
import utils, sys

opts = utils.parse(sys.argv[1:], {}, ".splunkrc")
if len(sys.argv) != 5:
    print "Syntax: mkuser.py username password role \"real name\""
    sys.exit()
    
try: 
    service = client.connect(**opts.kwargs)
    params = #LAB: create a name/value pair for the real name with value in sys.argv[4]

    user = #LAB: create a user with name in sys.argv[1], password in sys.argv[2], role in sys.argv[3], and pass params
    print "User %s created." % user.name

except Exception, err:
    sys.stderr.write('Error: %s\n' % str(err))

File: mkuser.java
import com.splunk.*; 
import java.util.*;
import java.io.*;

public class mkuser {

    public static void main(String[] args) {

		try {
			if(!(args.length == 4)) {
				System.out.println("syntax: mkuser username password role \"real name\"");
				System.exit(0);
			}
			Command command = Command.splunk("info").parse(args);
	        Service service = Service.connect(command.opts);

	        User user = //LAB: create the user named in args[0], password in args[1], and role in args[2]
	        //LAB: set the user's real name to args[3]
	        //LAB: send the user changes to the server
	        System.out.println("User " + user.getName() + " created.");
		} 
    	catch (Exception e)  {
    		if(e.getMessage() != null) {
    			System.err.println("Error message: " + e.getMessage());
    		} else {
    			System.err.println("Error type: " + e.getClass());
    		}
		}
	}
}